Sources/KituraContracts/Contracts.swift

```diff
@@ -523,6 +523,21 @@ extension RequestError {
  - All other non-optional types throw a decoding error
  */
 public protocol QueryParams: Codable {
+
+    static var dateDecoder: JSONDecoder.DateDecodingStrategy { get }
+    static var dateEncoder: JSONEncoder.DateEncodingStrategy { get }
```

Comment:
This doesn't play nicely with the values defined on `QueryEncoder` / `QueryDecoder`, as this will always override them.  You could either:
1) Make these properties optionals, and in the extension assign them a default value of `nil`.  When configuring the encoder/decoder during an operation, this becomes an override only if it is non-nil.
2) Leave these as-is, and make the `QueryEncoder.dateEncoder` / `QueryDecoder.dateDecoder` private.

Otherwise, this looks like it should work, but it won't:
```swift
struct MyParams: QueryParams {
    let date: Date
}
let params = MyParams(date: Date(timeIntervalSinceNow: 0))
let encoder = QueryEncoder()
encoder.dateEncoder = .iso8601   // Expect this to be used to encode the date
let data = try params.encode(to: encoder)    // Actually uses QueryParams' default dateEncoder
```

I think that defining the value on the struct makes more sense than defining it on the Encoder / Decoder itself, although that's what we do for uses of JSONEncoder/decoder in Codable routing.  Option 1 is more flexible in that either are permitted, so I'd opt for that approach.

Reply:
Also I'd name them `dateEncodingStrategy` / `dateDecodingStrategy`